core/src/main/java/org/apache/carbondata/core/memory/UnsafeMemoryManager.java

```diff
@@ -123,4 +121,29 @@ public boolean isMemoryAvailable() {
   public long getUsableMemory() {
     return totalMemory - minimumMemory;
   }
+
+  /**
+   * It tries to allocate memory of `size` bytes, keep retry until it allocates successfully.
+   */
+  public static MemoryBlock allocateMemoryBlocking(long size) throws MemoryException {
```

Comment:
Actually it is not a typo, I want to express this function will block (retry) until it allocate memory successfully. Since this name is not good, I will change it to `allocateMemoryWithRetry`